deltachat-ios/AppDelegate.swift

```diff
@@ -484,35 +484,33 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
         pushToDebugArray("1")
 
         // move work to non-main thread to not block UI (otherwise, in case we get suspended, the app is blocked totally)
-        // (we are using `qos: default` as `qos: .background` or `main.asyncAfter` may be delayed by tens of minutes)
+        // (we are using `qos: default` as `qos: .background` may be delayed by tens of minutes)
         DispatchQueue.global().async { [weak self] in
             guard let self else { completionHandler?(.failed); return }
 
-            // we're in background, run IO for a little time
-            self.dcAccounts.startIo()
-            self.dcAccounts.maybeNetwork()
             self.pushToDebugArray("2")
 
             self.addDebugFetchTimestamp()
-
-            // create a new semaphore to make sure the received DC_CONNECTIVITY_CONNECTED really belongs to maybeNetwork() from above
-            // (maybeNetwork() sets connectivity to DC_CONNECTIVITY_CONNECTING, when fetch is done, we're back at DC_CONNECTIVITY_CONNECTED)
             self.dcAccounts.fetchSemaphore = DispatchSemaphore(value: 0)
+            let start = CFAbsoluteTimeGetCurrent()
+
+            // backgroundFetch() pauses IO as needed
+            if !self.dcAccounts.backgroundFetch(timeout: 20) {
+                logger.error("backgroundFetch failed")
+                self.pushToDebugArray("ERR2")
+            }
+            let diff = CFAbsoluteTimeGetCurrent() - start
+
+            // wait for DC_EVENT_ACCOUNTS_BACKGROUND_FETCH_DONE,
+            // marking the end of events needed to being processed.
+            // as IO was not started, random events caused by other reasons are not added.
```

Comment:
maybe also mention that this event is also emitted in case of timeout, so it is clearer that we can not just do an early exit when the fetch timed-out (because the event is still emitted and we still might have a partial fetch that has some of the new message events)

Reply:
i made the comment more precise, putting it more in code context and not just repeating documentation

main purpose of the comment is anyway to know that you need to grep for DC_EVENT_ACCOUNTS_BACKGROUND_FETCH_DONE :)